src/main/java/org/parabot/api/ui/JavaFxUtil.java

```diff
@@ -0,0 +1,156 @@
+package org.parabot.api.ui;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import org.parabot.api.Configuration;
+import org.parabot.api.io.ProgressListener;
+import org.parabot.api.io.WebUtil;
+import org.parabot.api.output.Verboser;
+
+import javax.swing.*;
+import java.awt.event.WindowAdapter;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+/**
+ * If a JavaFX program runs as an "Application" it can only be launched once due to
+ * how the UI threadding works. To solve this, JavaFX should be used by embedded the FX content inside an FXPanel,
+ * which itself should just fill the content of a empty JFrame.
+ *
+ * @author Shadowrs
+ */
+public abstract class JavaFxUtil {
+
+    private       JFrame     frame;
+    private       JFXPanel   jfxp;
+    private       JavaFxUtil INSTANCE;
+    private final URL        end;
+    private final File       target;
+    private final Class<?> controller;
+    private final ProgressListener listener;
+
+    /**
+     * Constructor to use when stylesheet.fxml is in the Parabot Jar
+     * @param controllerPath Location to .fxml such as "/storage/ui/notifications.fxml"
+     */
+    public JavaFxUtil(final String controllerPath, final Class<?> controller) {
+        URL end1;
+        try {
+            end1 = Configuration.class.getClass().getResource(controllerPath).toURI().toURL();
```

Comment:
`this.end = Configuration.class.getClass().getResource(controllerPath).toURI().toURL();`